File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/trash/Patient.java
package gov.va.med.mhv.sm.model;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import gov.va.med.mhv.sm.enumeration.ParticipantTypeEnum;
import gov.va.med.mhv.sm.enumeration.UserTypeEnum;

/**
 * 
 * @author vhalommccarw
 *
 *  represents a specific actor of patient
 *  
 */
public class Patient extends User implements Serializable{

	
	/**
	 * 
	 */
	private static final long serialVersionUID = -8448442975344919006L;
	protected Date dob;
	protected String icn; 
	protected List<PatientFacility> facilities;	
	protected Date relationshipUpdate;

	public Patient(){
		super();
		this.userType = UserTypeEnum.PATIENT;
		this.participantType = ParticipantTypeEnum.PATIENT;
	}

	public Date getDob() {
		return dob;
	}
	public void setDob(Date dob) {
		this.dob = dob;
	}
	public String getIcn() {
		return icn;
	}
	public void setIcn(String icn) {
		this.icn = icn;
	}
	public List<PatientFacility> getFacilities() {
		return facilities;
	}
	public void setFacilities(List<PatientFacility> facilities) {
		this.facilities = facilities;
	}
	public Date getRelationshipUpdate() {
		return relationshipUpdate;
	}
	public void setRelationshipUpdate(Date relationshipUpdate) {
		this.relationshipUpdate = relationshipUpdate;
	}

}

File: OtherComponents/MDWSvistalayer/MDWS Source/mdo/mdo/src/mdo/domain/sm/Patient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using gov.va.medora.mdo.domain.sm.enums;

namespace gov.va.medora.mdo.domain.sm
{
    [Serializable]
    public class Patient : gov.va.medora.mdo.domain.sm.User
    {

        private DateTime _dob;

        public DateTime Dob
        {
            get { return _dob; }
            set { _dob = value; }
        }
        private string _icn;

        public string Icn
        {
            get { return _icn; }
            set { _icn = value; }
        }
        private List<PatientFacility> _facilities;

        public List<PatientFacility> Facilities
        {
            get { return _facilities; }
            set { _facilities = value; }
        }
        private DateTime _relationshipUpdate;

        public DateTime RelationshipUpdate
        {
            get { return _relationshipUpdate; }
            set { _relationshipUpdate = value; }
        }

        public Patient()
        {
            //UserType = UserTypeEnum.PATIENT;
            ParticipantType = ParticipantTypeEnum.PATIENT;
        }
    }
}
